### PR TITLE
docs/deploy add minikube and k3s deployment guides

### PR DESCRIPTION
**Pull request by abeatrix**
Close: https://github.com/sourcegraph/sourcegraph/issues/43574 & https://github.com/sourcegraph/sourcegraph/issues/43575

Add minikube and k3s deployment guides to docs:
- Users ([example](https://github.com/sourcegraph/sourcegraph/issues/34548)) have been using the minikube deployment guide that is currently available in our [handbook](https://handbook.sourcegraph.com/departments/ce-support/support/process/enablement/minikube/) but not available in our docs.
- K3s manual deployment instruction is also available in our [deploy repo ](https://github.com/sourcegraph/deploy/blob/main/doc/manual-ami.md) but not on docsite

It would be easier to consolidate them into one place

## Test plan

Docs update

image

### CODE REVIEW

**Review by jdpleiness — APPROVED** [review]
LGTM 👍

File: doc/admin/deploy/single-node/minikube.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+|----------------|-------------------------------|--------------|---------------------------|
+```
+
+That's it! You can now access the local Sourcegraph instance in browser using the URL or IP address retrieved from the previous steps 🎉
+
+<img class="screenshot" src="https://user-images.githubusercontent.com/68532117/141357183-905d0dbe-2d40-4dec-98b1-0a1cb13b0cf4.png" alt="minikube-startup-screen"/>
+
+## Upgrade
+
+Please refer to the [upgrade docs for all Sourcegraph kubernetes instances](../kubernetes/operations#upgrade).
```
Comment (jac) [inline doc/admin/deploy/single-node/minikube.md:141]: @abeatrix is this PR good to merge? Seems like all that is needed is this change for CI to pass
```suggestion
Please refer to the [upgrade docs for all Sourcegraph kubernetes instances](../kubernetes/operations.md#upgrade).
```

Comment (abeatrix) [reply doc/admin/deploy/single-node/minikube.md]: Thanks James! I wanted to make some changes to the current script but got sidetracked by other tasks, I can walkthrough the new script with y'all during pairing tomorrow before merging 😀

File: doc/index.md [hunk 2]
```diff
@@ -94,9 +94,19 @@ Sourcegraph is runnable in a variety of environments, from cloud to self-hosted
...
     <img alt="docker-logo" src="https://storage.googleapis.com/sourcegraph-resource-estimator/assets/Docker.png"/>
     <h3>Docker Container</h3>
-    <p>Spin up a local Sourcegraph instance</p>
+    <p>Spin up a local Sourcegraph instance with pure docker</p>
   </a>
-  <div></div><div></div>
+  <a class="btn-app btn" href="/admin/deploy/single-node/k3s">
+    <img alt="k3s-logo" src="/assets/other-logos/k3s.png"/>
+    <h3>K3s</h3>
+    <p>Spin up a local Sourcegraph instance with kubernetes</p>
```
Comment (jac) [inline doc/index.md:102]: nit:
```suggestion
    <p>Spin up a local Sourcegraph instance with Kubernetes</p>
```

File: doc/index.md [hunk 3]
```diff
@@ -94,9 +94,19 @@ Sourcegraph is runnable in a variety of environments, from cloud to self-hosted
...
+  </a>
+  <a class="btn-app btn" href="/admin/deploy/single-node/minikube">
+    <img alt="minikube-logo" src="/assets/other-logos/minikube.png"/>
+    <h3>Minikube</h3>
+    <p>Spin up a local Sourcegraph instance with kubernetes</p>
```
Comment (jac) [inline doc/index.md:107]: nit:
```suggestion
    <p>Spin up a local Sourcegraph instance with Kubernetes</p>
```

**Pull request merged**